flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosFlinkResourceManager.java

```diff
@@ -0,0 +1,755 @@
+import org.apache.flink.mesos.scheduler.messages.ReRegistered;
+import org.apache.flink.mesos.scheduler.messages.Registered;
+import org.apache.flink.mesos.scheduler.messages.ResourceOffers;
+import org.apache.flink.mesos.scheduler.messages.StatusUpdate;
+import org.apache.flink.mesos.util.MesosConfiguration;
+import org.apache.flink.runtime.clusterframework.ApplicationStatus;
+import org.apache.flink.runtime.clusterframework.FlinkResourceManager;
+import org.apache.flink.runtime.clusterframework.messages.FatalErrorOccurred;
+import org.apache.flink.runtime.clusterframework.messages.StopCluster;
+import org.apache.flink.runtime.clusterframework.types.ResourceID;
+import org.apache.flink.runtime.leaderretrieval.LeaderRetrievalService;
+import org.apache.mesos.Protos;
+import org.apache.mesos.Protos.FrameworkInfo;
+import org.apache.mesos.SchedulerDriver;
+import org.slf4j.Logger;
+import scala.Option;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Flink Resource Manager for Apache Mesos.
+ */
+public class MesosFlinkResourceManager extends FlinkResourceManager<RegisteredMesosWorkerNode> {
+
+	/** The Mesos configuration (master and framework info) */
+	private final MesosConfiguration mesosConfig;
+
+	/** The TaskManager container parameters (like container memory size) */
+	private final MesosTaskManagerParameters taskManagerParameters;
+
+	/** Context information used to start a TaskManager Java process */
+	private final Protos.TaskInfo.Builder taskManagerLaunchContext;
+
+	/** Number of failed Mesos tasks before stopping the application. -1 means infinite. */
+	private final int maxFailedTasks;
+
+	/** Callback handler for the asynchronous Mesos scheduler */
+	private SchedulerProxy schedulerCallbackHandler;
+
+	/** Mesos scheduler driver */
+	private SchedulerDriver schedulerDriver;
+
+	private ActorRef connectionMonitor;
+
+	private ActorRef taskRouter;
+
+	private ActorRef launchCoordinator;
+
+	private ActorRef reconciliationCoordinator;
+
+	private MesosWorkerStore workerStore;
+
+	final Map<ResourceID, MesosWorkerStore.Worker> workersInNew;
+	final Map<ResourceID, MesosWorkerStore.Worker> workersInLaunch;
+	final Map<ResourceID, MesosWorkerStore.Worker> workersBeingReturned;
+
+	/** The number of failed tasks since the master became active */
+	private int failedTasksSoFar;
+
+	public MesosFlinkResourceManager(
+		Configuration flinkConfig,
+		MesosConfiguration mesosConfig,
+		MesosWorkerStore workerStore,
+		LeaderRetrievalService leaderRetrievalService,
+		MesosTaskManagerParameters taskManagerParameters,
+		Protos.TaskInfo.Builder taskManagerLaunchContext,
+		int maxFailedTasks,
+		int numInitialTaskManagers) {
+
+		super(numInitialTaskManagers, flinkConfig, leaderRetrievalService);
+
+		this.mesosConfig = requireNonNull(mesosConfig);
```

Comment:
for the sake of uniformity you could also use `o.a.f.util.Preconditions.checkNotNull()`.